### PR TITLE
Add settings to choose between spaces and tabs, as well as indent size

### PR DESCRIPTION
**Pull request by James-Livesey**
This pull request intends to introduce new editing features to the Espruino Web IDE. The proposed version adds two settings which can be configured on the General tab of the IDE:
* Whether to use spaces or tab characters when indenting using the Tab key
* The number of space characters one indentation takes up (works regardless of whether the user is using spaces or tab characters, options are `1`, `2`, `4` and `8`)

This pull request hopes to solve the [spaces vs tabs war](https://softwareengineering.stackexchange.com/questions/57/tabs-versus-spaces-what-is-the-proper-indentation-character-for-everything-in-e) (I'm a 4-character spacer btw!) which will hopefully make everybody more at home in their IDE.

Also in this pull request:
* Some of the behaviour when you press the Tab key has changed to fix issues when the user chooses to indent using more than 2 spaces. It does away with the `cm.indentSelection("add");` command, but replicates its behaviour (including snapping to the next tab stop), sans the issues associated with it.

### CODE REVIEW

**Comment by James-Livesey** [comment]
**Important note:** If you're a maintainer, please ensure that you add this feature PR to any changelogs since I don't know if you have a changelog.

File: js/core/editorJavaScript.js [hunk 1]
```diff
@@ -47,6 +47,30 @@
...
+    });
+    Espruino.Core.Config.add("TAB_SIZE", {
+      section : "General",
+      name : "Indentation Size",
+      description : "The number of space characters an indentation should take up",
+      type : {1:1,2:2,4:4,8:8},
+      defaultValue : 2,
+      onChange : function(newValue) {
+        if (codeMirror) {
+          codeMirror.setOption('indentUnit', Espruino.Config.TAB_SIZE);
```
Comment (gfwilliams) [inline js/core/editorJavaScript.js:70]: Should this handler also set `tabSize` since that's set on line 106 as well?

**Comment by gfwilliams** [comment]
Great, thanks! Just posted a comment about `tabSize`. I don't suppose that could have been causing issues with `cm.indentSelection("add");` - I'm just super surprised that something like codemirror is incapable of indenting to 4 spaces.

**Comment by James-Livesey** [comment]
It's a strange bug and to be quite honest, I am too surprised that
CodeMirror doesn't seem to play well with 4-space indentation! Though I do
recall that I have previously got it to work with 4 spaces when I played
with CodeMirror myself a long time ago, but the configuration was probably
significantly more simple than what is implemented in the Espruino Web IDE.
I could always look at the code I wrote then to see what config I had and
try it in the code for this PR.

When I tried the indentSelection command in the Web IDE, the indentation
went crazy when you indent twice — it'd jump forwards about 40 spaces or
so... Strange!

File: js/core/editorJavaScript.js (hunk 1 above)
Comment (James-Livesey) [reply js/core/editorJavaScript.js:70]: Having read the code you highlighted, that could well be the issue... Strange how CodeMirror implements two separate properties when changing tab size! Will take a look tonight and see if that's the cause of the crazy indentation problems.

**Comment by gfwilliams** [comment]
Thanks! Just merged :)

**Comment by James-Livesey** [comment]
Awesome! Glad to see it as a new feature 🎉

**Pull request merged**